src/java.base/share/classes/java/lang/String.java

```diff
@@ -3126,13 +3126,50 @@ public String replace(CharSequence target, CharSequence replacement) {
      * Splits this string around matches of the given
      * <a href="../util/regex/Pattern.html#sum">regular expression</a>.
      *
-     * <p> The array returned by this method contains each substring of this
+     * <p> This method works as if by invoking the three-argument {@link
+     * #split(String, int, boolean) split} method with the given {@code regex},
+     * the given {@code limit}, and {@code false} as the last argument.
+     *
+     * @param  regex
+     *         the delimiting regular expression
+     *
+     * @param  limit
+     *         the result threshold, as described above
+     *
+     * @return  the array of strings computed by splitting this string
+     *          around matches of the given regular expression
+     *
+     * @throws  PatternSyntaxException
+     *          if the regular expression's syntax is invalid
+     *
+     * @see java.util.regex.Pattern
+     *
+     * @since 1.4
+     */
+    public String[] split(String regex, int limit) {
+        return split(regex, limit, false);
+    }
+
+    /**
+     * Splits this string around matches of the given
+     * <a href="../util/regex/Pattern.html#sum">regular expression</a>.
+     *
+     * <p> When {@code withDelimiter} is {@code false},
```

Comment:
I'd look for an brief but intuitive description of the difference between the two cases before diving into the details.
It might even be better to not expose the implementation with the true/false argument but have a separate `splitWithDelimiters` (name tbd) method that can avoid having to describe both cases.

I understand this something like:
   Splits this string around matches of the given regular expression and returns both the strings and the matching delimiters.